client/web/src/components/login-toggle.tsx

```diff
@@ -95,9 +95,16 @@ function LoginPopoverContent({
   const [canConnectOverTS, setCanConnectOverTS] = useState<boolean>(false)
   const [isRunningCheck, setIsRunningCheck] = useState<boolean>(false)
 
+  // Whether the current page is loaded over HTTPS.
+  // If it is, then the connectivity check to the management client
+  // will fail with a mixed-content error.
+  const isHTTPS = window.location.protocol === "https:"
+
   const checkTSConnection = useCallback(() => {
-    if (auth.viewerIdentity) {
-      setCanConnectOverTS(true) // already connected over ts
+    if (auth.viewerIdentity || isHTTPS) {
+      // Either Already connected over Tailscale, or we know the
+      // connectivity check will fail so just skip it.
```

Comment:
nit, including HTTPS explicitly in the comment
```suggestion
      // Skip the connectivity check if we either already know we're connected over Tailscale,
      // or know the connectivity check will fail because the current page is loaded over HTTPS.
```

Reply:
done.